docs/resource_sets/resource_sets.rst

```diff
@@ -0,0 +1,70 @@
+responsibility of the server to compose a new version of the model using the resources in the previous version of the
+model and the resources that are part of the partial compile.
+
+The example shown below will update the previous example with a partial model:
+
+.. literalinclude:: resource_sets_sources/basic_example_partial.cf
+    :language: inmanta
+    :caption: main.cf
+
+as a result, the id of Router b is changed, router c is removed and router z is added to the model. The equivalent full model would look like this:
+
+.. literalinclude:: resource_sets_sources/basic_example_full_result.cf
+    :language: inmanta
+    :caption: main.cf
+
+Constraints and rules
+************************
+
+To ensure partial compiles work correctly some constraints and rules where put in place:
+
+* A resource cannot be part of multiple resource sets at the same time.
+* A resource doesn't have to be assigned to a resource set.
+* Resources cannot migrate to a different resource set using a partial compile. This operation requires a full compile.
+* When a partial compile contains a resource that belongs to a certain resource set, the partial compile should contain all the resources for that specific resource set that should be present in the new version of the model.
+* A partial compile can never update/remove resources that were not assigned to a specific resource set (adding resources is allowed).
+* The new version of the model that results from a partial compile, should have a dependency graph that is closed (i.e. doesn't have any dangling dependencies).
+* A partial compile can update multiple resource sets at the same time.
+
+Exporting a partial model to the server
+******************************************************
+To export a partial model to the server, 2 options can be specified to the ``inmanta export`` command.
+
+- A ``--partial`` option is added to indicate that the model being compiled, only contains the resources that should be updated with respect to the previous version of the model.
+
+- A ``--delete-resource-set <resource-set-name>`` option is added. This option can be passed multiple times and indicates that the resource set with the given name should be removed from the model. This option can only be used in combination with the previous one. Note that it is also possible to implicitly delete resource sets in a partial compile by using an ``std::ResourceSet`` that doesn't contains any resources
+
+Limitations
+*************
+* With partial compiles the compiler cannot verify all constraints that would be checked when a full compile is ran. For example, not all index constraints can be verified. It's the responsibility of the model developer to make sure that these constraints are satisfied.
+* By only doing partial compile it can happen that a shared resource becomes obsolete: several resources that belongs to different resources set can depend on a specific shared resource (not associated with a specific resource set).
```

Comment:
```suggestion
* By only doing partial compile it can happen that a shared resource becomes obsolete: several resources that belong to a different resource set can depend on a specific shared resource (not associated with a specific resource set).
```